api/oauth2_integration_test.go

```diff
@@ -0,0 +1,581 @@
+//go:build integration
+// +build integration
+
+package api
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/frain-dev/convoy/api/models"
+	"github.com/frain-dev/convoy/api/testdb"
+	"github.com/frain-dev/convoy/cache"
+	mcache "github.com/frain-dev/convoy/cache/memory"
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/database"
+	"github.com/frain-dev/convoy/database/postgres"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/internal/pkg/keys"
+	"github.com/frain-dev/convoy/internal/pkg/metrics"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/services"
+	"github.com/oklog/ulid/v2"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type OAuth2IntegrationTestSuite struct {
+	suite.Suite
+	DB              database.Database
+	Router          http.Handler
+	ConvoyApp       *ApplicationHandler
+	AuthenticatorFn AuthenticatorFn
+	DefaultOrg      *datastore.Organisation
+	DefaultProject  *datastore.Project
+	DefaultUser     *datastore.User
+	OAuth2Server    *httptest.Server
+	WebhookServer   *httptest.Server
+	MemoryCache     cache.Cache
+	// Track requests to servers
+	OAuth2TokenRequests  []map[string]string
+	WebhookRequests      []http.Request
+	OAuth2TokenCallCount int
+	WebhookCallCount     int
+}
+
+func (s *OAuth2IntegrationTestSuite) SetupSuite() {
+	s.DB = getDB()
+	s.ConvoyApp = buildServer()
```

Comment:
Let's update the tests to use the new test setup. Btw once merged to main this line won't compile anymore `s.ConvoyApp = buildServer()`.